src/vm/memory/memory.zig

```diff
@@ -654,6 +691,11 @@ test "memory set and get" {
     );
     const value_2 = relocatable.fromFelt(starknet_felt.Felt252.one());
 
+    try memory.data.append(std.ArrayListUnmanaged(?MemoryCell){});
+    memory.num_segments += 1;
+    try memory.temp_data.append(std.ArrayListUnmanaged(?MemoryCell){});
+    memory.num_segments += 1;
```

Comment:
Here, you then use the `setUpMemory` function and it seems that you already take into account the addition of segments in this function, right? 

In this case there is no need to do it manually above, right? Which is preferable from a scalability point of view for the rest of our tests as well. We need to manage the addition of segments directly in the `setUpMemory` function to be able to setup our tests with as few manual procedures as possible.